org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala

```diff
@@ -371,57 +371,61 @@ trait ClasspathManagement extends HasLogger { self: ScalaProject =>
+
+    def singleLibFound(lib: ScalaLibrary) = lib.version match {
+      case Some(v) if mode == Same && v == ScalaPlugin().scalaVersion || ScalaPlugin().isCompatibleVersion(v, this) =>
+        Nil
+      case Some(v) if isBinaryPrevious(ScalaPlugin().scalaVersion, v) =>
+        val msg = s"The version of scala library found in the build path of ${underlying.getName()} (${v.unparse}) is prior to the one provided by scala IDE ($scalaVersion). Setting a Scala Installation Choice to match."
+        // It's important here to check we're not mistakenly "fixing" the scala installation of a project which already has a scala container on classpath
+        // Those should have their installation choice changed through other means, we only aim at changing installation for 'unmanaged' (non-container) libs, e.g. sbt imports
+        if (canFixInstallationFromScalaLib && !isBundledPath(fragmentRoots(0).location)) {
+          // see the comment to checkClasspath above
+          EclipseUtils.scheduleJob(s"Update Scala Installation from raw classpath for ${underlying.getName()}", underlying, Job.BUILD) { monitor =>
+            projectSpecificStorage.setValue(SettingConverterUtil.SCALA_DESIRED_INSTALLATION, ScalaInstallationChoice(v).toString())
+            setDesiredInstallation(ScalaInstallationChoice(v), "requested Scala Installation change from classpath analysis at project open")
+            projectSpecificStorage.save()
+            publish(ScalaInstallationChange())
+            Status.OK_STATUS
           }
-          case None =>
-            // no version found
-            ClasspathErrorMarker(IMarker.SEVERITY_ERROR, "The scala library found in the build path doesn't expose its version. Please replace the scala library with the scala container or a valid scala library jar", SdtConstants.ClasspathProblemMarkerId) :: Nil
+          ClasspathErrorMarker(IMarker.SEVERITY_WARNING, msg, SdtConstants.ScalaVersionProblemMarkerId) :: Nil
         }
-      case _ => // 2 or more of them, not great, but warn only if the library is not a project
-        if (fragmentRoots.exists(incompatibleScalaLibrary))
-          ClasspathErrorMarker(IMarker.SEVERITY_ERROR, moreThanOneLibraryError(fragmentRoots.map(_.location), compatible = false), SdtConstants.ClasspathProblemMarkerId) :: Nil
+        // Previous version, and the XSource flag isn't there already : warn and suggest fix using Xsource
+        else ClasspathErrorMarker(IMarker.SEVERITY_ERROR, msg, SdtConstants.ScalaVersionProblemMarkerId) :: Nil
+      case Some(v) =>
+        // incompatible version
+        ClasspathErrorMarker(IMarker.SEVERITY_ERROR, s"The version of scala library found in the build path (${v.unparse}) is incompatible with the one expected by scala IDE ($expectedVersion). Please replace the scala library with the scala container or a compatible scala library jar.", SdtConstants.ClasspathProblemMarkerId) :: Nil
```

Comment:
Shouldn't this allow a `Subsequent` libraries? If there's only one scala lib, 2.12, will it work?